core/src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorSpec.scala

```diff
@@ -694,6 +694,87 @@ class SubmissionMonitorSpec(_system: ActorSystem) extends TestKit(_system) with
     }
   }
 
+  it should "fail workflows that exceed the configured workspace attribute maximum" in withDefaultTestDatabase { dataSource: SlickDataSource =>
+
+    runAndWait {
+      withWorkspaceContext(testData.workspace) { context =>
+        submissionQuery.create(context, testData.submissionMaxWorkspaceAttributes)
+      }
+    }
+
+    val monitor = createSubmissionMonitor(dataSource, mockSamDAO, mockGoogleServicesDAO, testData.submissionMaxWorkspaceAttributes, testData.wsName, new SubmissionTestExecutionServiceDAO(WorkflowStatuses.Succeeded.toString))
+    val workflowRecs = runAndWait(workflowQuery.listWorkflowRecsForSubmission(UUID.fromString(testData.submissionMaxWorkspaceAttributes.submissionId)))
+
+    runAndWait(monitor.handleOutputs(workflowRecs.map(r => (r, ExecutionServiceOutputs(r.externalId.get, Map("o1" -> Left(
+      AttributeValueList(
+        Vector(
+          AttributeString("entry 1"),
+          AttributeString("entry 2"),
+          AttributeString("entry 3"),
+          AttributeString("entry 4"),
+          AttributeString("entry 5"),
+          AttributeString("entry 6"),
+          AttributeString("entry 7"),
+          AttributeString("entry 8"),
+          AttributeString("entry 9"),
+          AttributeString("entry 10"),
+          AttributeString("entry 11")
+    ))))))), this))
+
+    val workflowRecord = runAndWait(
+      workflowQuery.findWorkflowByExternalIdAndSubmissionId(
+        testData.submissionMaxWorkspaceAttributes.workflows.head.workflowId.get,
+        UUID.fromString(testData.submissionMaxWorkspaceAttributes.submissionId)).result).head
+
+    val errorMessage = "Cannot save outputs to workspace because workflow's attribute count of 15 exceeds Terra maximum of 10."
```

Comment:
for my edification: why is this 15 when I only see 11 entries just above it at line 708?

Reply:
I agree it's a bit confusing, these tests share a workspace that already has some attributes: https://github.com/broadinstitute/rawls/blob/53b88c392dd779bbe805683eae79853aaefd20a1/core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/TestDriverComponent.scala#L255-L260

Reply:
does this mean that when a workflow writes to workspace attributes, all pre-existing attributes are rewritten even if they haven't changed? If that's the case, we should fix that (obviously not for this PR)

Reply:
Looks like it, yeah: https://github.com/broadinstitute/rawls/blob/81d6a688a77c90489e1e3a0b3377b4015e358122/core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala#L504-L512